### ISSUE TITLE
[Task]: Refactor and revist the useClick hook

### ISSUE THREAD

**Issue by SatishGandham** [appsmithorg/appsmith#20177]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

We have a delay of 200ms before we pass the click event to the handler. Removing this will give a very noticeable improvement in UX for the users.
https://github.com/appsmithorg/appsmith/blob/release/app/client/src/utils/hooks/useClick.tsx#L25

**Comment by ArjobanSingh** [comment]
You are right, but as far as I can understand, this timeout is probably to handle doubleClick, so if the developer has not passed the doubleClick handler, the single click would be handled instantly. It would just add delay only if the developer has given the doubleClick function, but the user clicked only once and the developer also wants to add single-click support on that particular element.

**Comment by SatishGandham** [comment]
@ArjobanSingh, My expectation was to find an alternate way to implement double click, but I did not notice the condition. 
Thank you :)

If think the problem would be resolved if we can pass undefined or skip the third argument.
https://github.com/appsmithorg/appsmith/blob/release/app/client/src/pages/Editor/Explorer/Entity/index.tsx#L301

**Comment by ArjobanSingh** [comment]
@SatishGandham yes in the referenced code you shared, we do not need to pass `noop` as the third argument, as it delays the click event. If you want, I can raise a PR, to fix this.

**Comment by alex-golovanov** [comment]
I'd like to share my observations, after looking into the issue:
1. We cannot remove the delay, because it is inherent to differentiating between single and double clicks. Both events are attached to the same element: in case of single click we navigate, while double click starts edit/rename mode for that item.
2. Composition is very confusing. `useClick` is being used in a parent component, while the real handler is attached in a child component through `onDoubleClick` prop. It would make sense to move useClick directly to `EntityName` component.

Bottom line, the delay cannot be removed, but components can be refactored to improve the composition.

@KelvinOm I propose we close this issue, because the original request of performance improvement cannot be completed.
